src/lib/phdi-building-blocks/phdi_building_blocks/geo.py

```diff
@@ -1,119 +1,193 @@
+    # Default assumption is addresses are same, in case there's no
+    # valid geocoder result
+    address_is_different = False
+    if geocoded_result:
+        std_one_line = f"{geocoded_result.address} {geocoded_result.city}, {geocoded_result.state} {geocoded_result.zipcode}"  # noqa
+        address_is_different = raw_one_line != std_one_line
 
-        raw_addresses = []
-        std_addresses = []
-        for address in patient.get("address", []):
-            # Generate a one-line address to pass to the geocoder
-            one_line = " ".join(address.get("line", []))
-            one_line += f" {address.get('city')}, {address.get('state')}"
-            if "postalCode" in address and address["postalCode"]:
-                one_line += f" {address['postalCode']}"
-            raw_addresses.append(one_line)
-
-            geocoded = geocode(client, one_line)
-            std_one_line = ""
-            if geocoded:
-                address["line"] = geocoded.address
-                address["city"] = geocoded.city
-                address["state"] = geocoded.state
-                address["postalCode"] = geocoded.zipcode
-                std_one_line = f"{geocoded.address} {geocoded.city}, {geocoded.state} {geocoded.zipcode}"  # noqa
-                std_addresses.append(std_one_line)
-
-                if "extension" not in address:
-                    address["extension"] = []
-
-                address["extension"].append(
-                    {
-                        "url": "http://hl7.org/fhir/StructureDefinition/geolocation",
-                        "extension": [
-                            {"url": "latitude", "valueDecimal": geocoded.lat},
-                            {"url": "longitude", "valueDecimal": geocoded.lng},
-                        ],
-                    }
-                )
-        any_dffs = (len(raw_addresses) != len(std_addresses)) or any(
-            [raw_addresses[i] != std_addresses[i] for i in range(len(raw_addresses))]
+    return (geocoded_result, address_is_different)
+
+
+def standardize_addresses_for_patient(
+    patient: dict, client: us_street.Client, add_metrics_extension: bool = True
+) -> None:
+    """
+    Helper function to handle the standardizing of all addresses belonging to
+    a single patient in a FHIR resource bundle. This function also determines
+    whether any of that patient's addresses were improved through geocoding and
+    tracks this in an extension on the patient profile.
+    :param dict patient: A Patient resource FHIR profile
+    :param us_street.Client client: The API client to geocode with
+    :param bool add_metrics_extension: Whether to add tracking metrics into the
+    patient resource denoting that one or more addresses was standardized.
+    :return: None
+    """
+
+    # We can track whether geocoding improved addresses by taking the logical
+    # OR everytime we geocode a new address
+    any_address_was_changed = False
```

Comment:
I know we talked about adding in these "trackers" for the purposes of demonstrating if our functions were having an impact on the data, but I don't think it should be part of our production code. In reality, we should have a notebook that reads a sample of files from the raw storage container, runs the data through our transformations using our SDK, and checks to see if the values changed. This should _not_ be part of production code. 

Happy to hear dissenting opinions.